### ISSUE TITLE
web3 1.0.0-beta.54 does not build anymore on new versions of node

### ISSUE THREAD

**Issue by lukebuehler** [urbit/azimuth-js#79]
On never version of node (>10), the web3 dependency does not build anymore because scrypt depends on node-gyp, which is not supported anymore.

Using azimuth-js in a project or even cloning this repo and doing `npm install` gives the following error:
```
> scrypt@6.0.3 install /home/me/dev/azimuth-js/node_modules/scrypt
> node-gyp rebuild

make: Entering directory '/home/me/dev/azimuth-js/node_modules/scrypt/build'
  SOLINK_MODULE(target) Release/obj.target/copied_files.node
  COPY Release/copied_files.node
  CC(target) Release/obj.target/scrypt_wrapper/src/util/memlimit.o
<....>
  CC(target) Release/obj.target/scrypt_lib/scrypt/scrypt-1.2.0/libcperciva/alg/sha256.o
  CC(target) Release/obj.target/scrypt_lib/scrypt/scrypt-1.2.0/libcperciva/util/insecure_memzero.o
  CC(target) Release/obj.target/scrypt_lib/scrypt/scrypt-1.2.0/lib/scryptenc/scryptenc_cpuperf.o
  AR(target) Release/obj.target/scrypt_lib.a
  COPY Release/scrypt_lib.a
  CXX(target) Release/obj.target/scrypt/src/node-boilerplate/scrypt_common.o
  CXX(target) Release/obj.target/scrypt/src/node-boilerplate/scrypt_params_async.o
In file included from ../src/node-boilerplate/inc/scrypt_async.h:28,
                 from ../src/node-boilerplate/inc/scrypt_params_async.h:28,
                 from ../src/node-boilerplate/scrypt_params_async.cc:4:
../src/node-boilerplate/inc/scrypt_common.h: In constructor ‘NodeScrypt::Params::Params(const v8::Local<v8::Object>&)’:
../src/node-boilerplate/inc/scrypt_common.h:39:48: error: no matching function for call to ‘v8::Object::Get(v8::Local<v8::String>)’
   39 |       N(obj->Get(Nan::New("N").ToLocalChecked())->Uint32Value()),
```

This is a know issue, See for example: https://github.com/axic/scrypt.js/issues/5

Updating web3 should fix this. Is there a reason we are still on 1.0.0-beta.54?

**Comment by jtobin** [comment]
> Updating web3 should fix this. Is there a reason we are still on 1.0.0-beta.54?

I've tried to update it before but have run into problems, e.g. #62.  Haven't had a chance to put a ton of work into it, and AFAIK it just hasn't been a major issue downstream yet.  Would certainly like to see it updated.

I can give it another go when I finish some other work, but if anyone would like to take a stab in the meantime, by all means, by my guest!

cc @Fang-

**Comment by lukebuehler** [comment]
I have a version of azimuth (azimuth-solidity) and azimuth-js working locally with all dependencies updated to the latest version, but I have not resolved the said (#62) test issues. I need it as part of my work on the azimuth-cli grant. The updates will certainly be needed when we release the cli because more people will npm install azimuth-js as a dependency.

Once I get closer to releasing the cli, I can also do PRs for updating this project and azimuth-solidity. But with all this, the greatest unknown will be how it affects bridge and your release pipeline for it.
